test/e2e/framework/util.go

```diff
@@ -416,29 +410,12 @@ func CheckTestingNSDeletedExcept(ctx context.Context, c clientset.Interface, ski
 	return fmt.Errorf("Waiting for terminating namespaces to be deleted timed out")
 }
 
-// WaitForServiceEndpointsNum waits until the amount of endpoints that implement service to expectNum.
-// Some components use EndpointSlices other Endpoints, we must verify that both objects meet the requirements.
+// WaitForServiceEndpointsNum waits until there are EndpointSlices for serviceName
+// containing a total of expectNum endpoints. (If the service is dual-stack, expectNum
+// must count the endpoints of both IP families.)
```

Comment:
This broke dual stack tests 

> I0121 11:20:30.804569 71111 util.go:431] Unexpected number of Endpoints on Slices, got 4, expected 2
I0121 11:20:31.782539 71111 util.go:418] Waiting for amount of service:node-port-service endpoints to be 2
I0121 11:20:31.792230 71111 util.go:431] Unexpected number of Endpoints on Slices, got 4, expected 2

https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-kubernetes-kind-network-dual/1881658530616840192